runtime/doc/lua.txt

```diff
@@ -1504,6 +1504,40 @@ schedule_wrap({cb})                                      *vim.schedule_wrap()*
         |vim.in_fast_event()|
 
 
+==============================================================================
+Lua module: inspector                                          *lua-inspector*
+
+inspect_pos({bufnr}, {row}, {col}, {filter})               *vim.inspect_pos()*
+    Get all the items at a given buffer position Can also be pretty printed
+    with *:Inspect!*
+
+    Parameters: ~
+      • {bufnr}   buffer|nil defaults to the current buffer
+      • {row}     (number|nil) 0-indexed row. Defaults to the current cursor
+      • {col}     (number|nil) 0-indexed col. Defaults to the current cursor
+      • {filter}  (table|nil) a table with key-value pairs to filter the items
+                  • syntax (boolean): include syntax based highlight groups
+                    (defaults to true)
+                  • treesitter (boolean): include treesitter based highlight
+                    groups (defaults to true)
+                  • extmarks (boolean): include extmarks (defaults to true)
+                  • extmarks_with_hl_group (boolean): when false, also include
```

Comment:
I would combine these keys and make the option an enum. Something like:

```suggestion
                  • extmarks (string): include extmarks. One of `all`, `none`, `with_hl`.
```

Reply:
good idea. Changed to be `true|false|"all"`, to keep it more in line with the other options.